helm/cluster-autoscaler-app/templates/deployment.yaml

```
@@ -23,12 +23,14 @@ spec:
       - effect: NoSchedule
         operator: "Exists"
         key: node-role.kubernetes.io/master
+      {{- if ((.Values.node).nodeSelector) }}
       nodeSelector:
-{{- if semverCompare ">=1.24.0" .Capabilities.KubeVersion.Version }}
-        node-role.kubernetes.io/control-plane: ""
-{{- else }}
-        node-role.kubernetes.io/master: ""
-{{- end }}
+        {{- if and (hasKey .Values.node.nodeSelector "node-role.kubernetes.io/control-plane") (semverCompare "<1.24.0" .Capabilities.KubeVersion.Version) }}
+        {{- toYaml (set (unset .Values.node.nodeSelector "node-role.kubernetes.io/control-plane") "node-role.kubernetes.io/master" "" )| nindent 8 }}
```

```suggestion
        {{- toYaml (set (unset .Values.node.nodeSelector "node-role.kubernetes.io/control-plane") "node-role.kubernetes.io/master" "" ) | nindent 8 }}
```

This way we are also dropping the value of the `node-role.kubernetes.io/control-plane`, but I don't think that really makes a difference... :)